Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/CreditCardBottomSheetViewModel.swift

```diff
@@ -146,26 +146,72 @@ struct SingleCreditCardViewModel {
     }
 
     // MARK: Helper Methods
-    func getPlainCreditCardValues() -> UnencryptedCreditCardFields? {
-        switch state {
+    func getPlainCreditCardValues(bottomSheetState: CreditCardBottomSheetState) -> UnencryptedCreditCardFields? {
+        switch bottomSheetState {
         case .save:
             guard let plainCard = decryptedCreditCard else { return nil }
             return plainCard
         case .update:
             guard let creditCard = creditCard,
-                  let ccNumberDecrypted = autofill.decryptCreditCardNumber(encryptedCCNum: creditCard.ccNumberEnc) else {
+                  let ccNumberDecrypted = autofill.decryptCreditCardNumber(encryptedCCNum: creditCard.ccNumberEnc)
+            else {
                 return nil
             }
+            let updatedDecryptedCreditCard = updateDecryptedCreditCard(from: creditCard,
+                                                                       with: ccNumberDecrypted,
+                                                                       fieldValues: decryptedCreditCard)
+            return updatedDecryptedCreditCard
+        }
+    }
 
-            let plainCard = UnencryptedCreditCardFields(
-                ccName: creditCard.ccName,
-                ccNumber: ccNumberDecrypted,
-                ccNumberLast4: creditCard.ccNumberLast4,
-                ccExpMonth: creditCard.ccExpMonth,
-                ccExpYear: creditCard.ccExpYear,
-                ccType: creditCard.ccType)
+    func getConvertedCreditCardValues(bottomSheetState: CreditCardBottomSheetState) -> CreditCard? {
```

Comment:
The only note if is we could add some this test for this func 🙏

Reply:
@yoanarios added a test for this as well, sorry if the test seem tedious as `enc card` are special case https://github.com/mozilla-mobile/firefox-ios/pull/14912/commits/3582d405f45e4a00f9c2dd34f29f95c2560b38a2

Reply:
Hoping BR gives no issues 🤞🏼 otherwise will have to disable one of the test :(